Utils/Piece.py

```diff
@@ -13,24 +13,29 @@ def get_picture(self):
         return self.picture
     def get_last_move(self):
         return self.last_move
+    
+    def set_position(self, x, y):
+        self.position = (y, x)
+        self.x = x
+        self.y = y
     def move(self, new_position):
-        self.position = new_position
+        self.set_position(new_position[1], new_position[0])
```

Comment:
As above - I would create some general type `Position` and utilize it everywhere